biosys/apps/main/api/views.py

```diff
@@ -350,6 +349,26 @@ def get_dataset(self, request, *args, **kwargs):
                 ds = get_object_or_404(Dataset, name=name)
         return ds
 
+    def get_queryset(self):
+        queryset = super(RecordViewSet, self).get_queryset()
+        if self.dataset:
+            # add some specific json field queries (postgres)
+            search_param = self.request.query_params.get('search')
+            if search_param is not None:
+                field_info = {
+                    'data': self.dataset.schema.field_names,
+                    'source_info': ['file_name', 'row']
+                }
+
+                queryset = search_json_fields(queryset, field_info, search_param)
+
+            ordering_param = self.request.query_params.get('ordering')
+            if ordering_param is not None:
+                queryset = order_by_json_field(queryset, 'data', self.dataset.schema.field_names, ordering_param)
+                queryset = order_by_json_field(queryset, 'source_info', ['file_name', 'row'], ordering_param)
```

Comment:
perhaps do a check that the ordering_param is 'file_name" or 'row' before calling this function.

Reply:
I think that the ordering parameter should be of form: `source_info__row` or `data__col`